addons/test_mail/tests/test_mail_multicompany.py

```diff
@@ -388,6 +389,43 @@ def _mock_filter_access_rules(*args, **kwargs):
                     other_activity_group = next(ag for ag in activity_groups if ag['model'] == 'mail.activity')
                     self.assertEqual(other_activity_group["total_count"], 5)
 
+    def test_cross_company_partner_mention(self):
+        """Test mentioning a partner with no common company."""
+        current_company, target_company = self.env["res.company"].create(
+            [{"name": "Current Company"}, {"name": "Target Company"}]
+        )
+        current_user, target_user = self.env["res.users"].create(
+            [
+                {
+                    "company_ids": [Command.link(current_company.id)],
+                    "company_id": current_company.id,
+                    "email": "current@example.com",
+                    "login": "Current User",
+                    "name": "Current User",
+                    "notification_type": "inbox",
+                },
+                {
+                    "company_ids": [Command.link(target_company.id)],
+                    "company_id": target_company.id,
+                    "email": "target@example.com",
+                    "login": "Target User",
+                    "name": "Target User",
+                    "notification_type": "inbox",
+                },
+            ]
+        )
+        thread = self.env["res.partner"].create({"name": "Test Partner"})
```

Comment:
Use test records, not partners as they are business records with specific behavior, MC rules, ....